internal/kvstore/kvstore.go

```diff
@@ -67,9 +71,14 @@ func (k *levelDBKeyValueStore) Delete(key string) error {
 	return err
 }
 
-func (k *levelDBKeyValueStore) NewIterator() KVIterator {
+func (k *levelDBKeyValueStore) NewIterator(slice ...interface{}) KVIterator {
```

Comment:
Might be clearer to have a separate `NewIteratorRange` function, if that's a small change before we merge.